abcpy/modelselections.py

```diff
@@ -0,0 +1,259 @@
+        n_samples_per_param : integer, optional 
+            Number of data points in each simulated data set. The default value is 1.
+        N_tree : integer, optional
+            Number of trees in the random forest. The default value is 100.
+        n_try_fraction : float, optional 
+            The fraction of number of summary statistics to be considered as the size of 
+            the number of covariates randomly sampled at each node by the randomised CART.
+            The default value is 0.5.
+        Returns
+        -------
+        abcpy.models.Model
+            A model which are of type abcpy.models.Model
+            
+        """        
+        # Creation of reference table
+        if self.reference_table_calculated is 0:        
+            rc = _RemoteContextgeneratereferencetable(self.backend, self.model_array, self.statistics_calc, observations, n_samples_per_param)
+
+            # Simulating the data, distance and statistics                
+            seed_arr = self.rng.randint(1, n_samples*n_samples, size=n_samples, dtype=np.int32)
+            seed_pds = self.backend.parallelize(seed_arr)     
+
+            model_data_pds = self.backend.map(rc._simulate_model_data, seed_pds)
+            model_data = self.backend.collect(model_data_pds)
+            models, data, statistics = [list(t) for t in zip(*model_data)]
+            self.reference_table_models = models
+            self.reference_table_data = data
+            self.reference_table_statistics = np.concatenate(statistics)
+            self.reference_table_calculated = 1
+        
+        # Construct a label for the model_array
+        label = np.zeros(shape=(len(self.reference_table_models)))
+        for ind1 in range(len(self.reference_table_models)):  
+            for ind2 in range(len(self.model_array)):
+                if self.reference_table_models[ind1] == self.model_array[ind2]:
+                    label[ind1] = ind2 
+
+        # Define the classifier 
+        classifier = RandomForestClassifier(n_estimators = N_tree, max_features=int(n_try_fraction*self.reference_table_statistics.shape[1]), bootstrap=True, random_state=self.seed)
+        classifier.fit(self.reference_table_statistics, label)  
+
+        pred_error = np.zeros(len(self.reference_table_models),)
+        # Compute missclassification error rate
+        for ind in range(len(self.reference_table_models)):
+            pred_error[ind] = 1 - classifier.predict_proba(self.statistics_calc.statistics(self.reference_table_data[ind]))[0][int(label[ind])] 
+
+        # Estimate a regression function with prediction error as response on summary statitistics of the reference table                
+        regressor = RandomForestRegressor(n_estimators = N_tree)
+        regressor.fit(self.reference_table_statistics,pred_error)
+
+        return(1-regressor.predict(self.statistics_calc.statistics(observations)))
+        
+class _RemoteContextgeneratereferencetable:
```

Comment:
Please use CamelCase to increase readability: _RemoteContextGenerateReferenceTable, maybe  _RemoteContextForReferenceTable is even clearer?